File: src/main/java/com/example/formatter/GoodDateFormatter.java
import java.text.DateFormat;
import java.util.Date;
import java.util.Locale;

public class DateFormatter {
    private final Locale locale;
    
    public DateFormatter(Locale locale) {
        this.locale = locale != null ? locale : Locale.getDefault();
    }
    
    public String formatDate(Date date) {
        if (date == null) {
            throw new IllegalArgumentException("Date cannot be null");
        }
        DateFormat df = DateFormat.getDateInstance(DateFormat.SHORT, locale);
        return df.format(date);
    }
    
    public Locale getCurrentLocale() {
        return locale;
    }
}

File: good_date_formatter.py
from datetime import datetime
import locale

class DateFormatter:
    def format_date(self, date: datetime, user_locale: str = 'en_US') -> str:
        try:
            locale.setlocale(locale.LC_TIME, user_locale)
            return date.strftime("%x")
        finally:
            locale.setlocale(locale.LC_TIME, '')